### PR TITLE
Refactor VulkanContext/Driver initialization

### PR DESCRIPTION
**Pull request by poweifeng**
Move the initialization of VulkanContext fields to the class itself. This will clean up how the various extension support flags are deteremined.

Also disable debugUtils extension for Mesa driver (#6192).

### CODE REVIEW

**Comment by poweifeng** [comment]
Sorry for the size of the PR. It's hard to break it apart. 
Please let me know if you prefer breaking it into more gradual chunks. Thanks.

File: filament/backend/src/vulkan/VulkanContext.h [hunk 1]
```diff
@@ -62,13 +62,17 @@ struct VulkanRenderPass {
...
 struct VulkanContext {
-    void selectPhysicalDevice();
-    void createLogicalDevice();
-    uint32_t selectMemoryType(uint32_t flags, VkFlags reqs);
-    VkFormat findSupportedFormat(utils::Slice<VkFormat> candidates, VkImageTiling tiling,
-            VkFormatFeatureFlags features);
+public:
+    void initialize(const char* const* ppRequiredExtensions, uint32_t requiredExtensionCount);
     void createEmptyTexture(VulkanStagePool& stagePool);
+    uint32_t selectMemoryType(uint32_t flags, VkFlags reqs);
```
Comment (bejado) [inline filament/backend/src/vulkan/VulkanContext.h:68]:
```suggestion
    uint32_t selectMemoryType(uint32_t flags, VkFlags reqs);
```

File: filament/backend/src/vulkan/VulkanContext.h [hunk 2]
```diff
@@ -62,13 +62,17 @@ struct VulkanRenderPass {
...
+public:
+    void initialize(const char* const* ppRequiredExtensions, uint32_t requiredExtensionCount);
     void createEmptyTexture(VulkanStagePool& stagePool);
+    uint32_t selectMemoryType(uint32_t flags, VkFlags reqs);    
 
+private:
+    void afterSelectPhysicalDevice();
+    void afterCreateLogicalDevice();
+    void afterCreateInstance();
+
```
Comment (bejado) [inline filament/backend/src/vulkan/VulkanContext.h:74]:
```suggestion
 
```

File: filament/backend/src/vulkan/VulkanContext.h [hunk 3]
```diff
@@ -93,6 +97,9 @@ struct VulkanContext {
     VmaAllocator allocator;
     VulkanTexture* emptyTexture = nullptr;
     VulkanCommands* commands = nullptr;
+    VkDebugReportCallbackEXT debugCallback = VK_NULL_HANDLE;
+    VkDebugUtilsMessengerEXT debugMessenger = VK_NULL_HANDLE;
+
```
Comment (bejado) [inline filament/backend/src/vulkan/VulkanContext.h:102]:
```suggestion
 
```

File: filament/backend/src/vulkan/VulkanContext.h [hunk 4]
```diff
@@ -62,13 +62,17 @@ struct VulkanRenderPass {
...
+    void initialize(const char* const* ppRequiredExtensions, uint32_t requiredExtensionCount);
     void createEmptyTexture(VulkanStagePool& stagePool);
+    uint32_t selectMemoryType(uint32_t flags, VkFlags reqs);    
 
+private:
+    void afterSelectPhysicalDevice();
+    void afterCreateLogicalDevice();
+    void afterCreateInstance();
+    
+public:
```
Comment (bejado) [inline filament/backend/src/vulkan/VulkanContext.h:75]:
```suggestion
public:
```

**Review by bejado — APPROVED** [review]
Just some trailing whitespace nits! Otherwise LGTM

File: filament/backend/src/vulkan/VulkanContext.h (hunk 1 above)
Comment (poweifeng) [reply filament/backend/src/vulkan/VulkanContext.h]: done

File: filament/backend/src/vulkan/VulkanContext.h (hunk 3 above)
Comment (poweifeng) [reply filament/backend/src/vulkan/VulkanContext.h]: done

File: filament/backend/src/vulkan/VulkanContext.h (hunk 2 above)
Comment (poweifeng) [reply filament/backend/src/vulkan/VulkanContext.h]: done

File: filament/backend/src/vulkan/VulkanContext.h (hunk 4 above)
Comment (poweifeng) [reply filament/backend/src/vulkan/VulkanContext.h]: done

**Pull request merged**